js/src/modals/Verification/email-store.js

```diff
@@ -68,8 +70,53 @@ export default class EmailVerificationStore extends VerificationStore {
     return hasReceivedCode(this.email, this.account, this.isTestnet);
   }
 
+  // If the email has already been used for verification of another account,
+  // we prevent the user from wasting ETH to request another verification.
+  @action checkIfAbleToRequest = () => {
+    const { isEmailValid } = this;
+
+    if (!isEmailValid) {
+      this.isAbleToRequest = true;
+      return;
+    }
+
+    const { contract, email } = this;
+    const emailHash = sha3.text(email);
+
+    this.isAbleToRequest = null;
+    contract
+      .instance.reverse
+      .call({}, [ emailHash ])
+      .then((address) => {
+        if (address === ZERO20) {
+          this.isAbleToRequest = true;
+        } else {
+          this.isAbleToRequest = new Error('Another account has been verified using this e-mail.');
```

Comment:
Would have preferred another way instead of a bool + Error type mixed. (e.g. set to bool in previous statement, as Error here)